### PR TITLE
docs: update docs for List, Modal, Pagination and popover component

### PR DESCRIPTION
**Pull request by Annaleah123**
### PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] Tests for the changes have been added (for bug fixes/features)
-   [x] Docs have been added/updated (for bug fixes/features)

### PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Performance improves
-   [ ] Adding missing tests
-   [x] Documentation content changes
-   [ ] Other... Please describe:

### Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### Additional context

Provide any additional context or screenshots about the feature PR.

### CODE REVIEW

File: packages/core/src/components/Modal/types.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface ModalProps extends HTMLProps<HTMLDivElement> {
     minWidth?: string;
     /** Min height in px/rem/% (1rem = 10px) */
     minHeight?: string;
-    /** Set it true to close the modal on click outside of popup */
+    /** Set it true to close the modal when clicking outside the popup */
```
Comment (abhishekashyap) [inline packages/core/src/components/Modal/types.ts:16]:
```suggestion
    /** Set it true to close the modal when clicked outside the modal */
```

File: packages/core/src/components/Modal/Modal.stories.mdx [hunk 2]
```diff
@@ -6,9 +6,8 @@ import * as stories from './Modal.stories';
 
 # Modal
 
-Basic modal component. You can play with the component in the canvas tab.
-
-Modal consists of three sections Header, Content and Actions. All of these you can directly use from Modal component only like bellow . . .
+The `Modal` components are separate windows within an application that are mostly overlaid on the primary window. The contents behind a modal dialog are inert, meaning you cannot interact with the content behind the dialog. It is a common user interface pattern that provides information or requires confirmation.
```
Comment (gmukul01) [inline packages/core/src/components/Modal/Modal.stories.mdx:9]: We should change it to `component is separate windows within an application that is overlaid on the primary window`

File: packages/core/src/components/Modal/Modal.stories.mdx [hunk 3]
```diff
@@ -41,28 +40,30 @@ DummyComponent.displayName = 'DummyComponent';
...
+The shadow effect apears on the Header & Actions section if you scroll through the content.
 
 ### Keyboard Support
 
-It also supports keyboard navigation when it is in focus
-
-| Key | Function        |
-| --- | --------------- |
-| Esc | Close the modal |
+The Modal component supports keyboard navigation when in focus. You can use the Esc key to close the modal.
```
Comment (gmukul01) [inline packages/core/src/components/Modal/Modal.stories.mdx:47]: Could we please add the table back to maintain the consistency across the components.

File: packages/core/src/components/Popover/Popover.stories.mdx [hunk 4]
```diff
@@ -82,19 +88,21 @@ The popup can be used with/without an arrow using the `withArrow` prop. The arro
...
+Use the Popover Context as mentioned below:
 
 ```tsx
 const [isOpen, setOpenState] = useContext(Popover.Context);
 ```
 
-Type of Popover Context is:
+## Type
+
+Use the Popover Context type as mentioned below:
```
Comment (gmukul01) [inline packages/core/src/components/Popover/Popover.stories.mdx:105]: The below type is just for information not for use.

**Review by gmukul01 — CHANGES_REQUESTED** [review]

**Review by gmukul01 — APPROVED** [review]
LGTM

**Pull request merged**